controllers/user.js

```diff
@@ -13,3 +16,85 @@ exports.findAll = (req, res) => {
       })
     })
 }
+
+exports.registerUser = async (req, res) => {
+  // validation with express-validator
+  const errors = validationResult(req)
+  if (!errors.isEmpty()) {
+    return res.status(400).json({
+      ok: false,
+      msg: 'Validation error',
+      error: errors.array(),
+    })
+  }
+
+  // hash the password
+  const salt = await bcrypt.genSalt(10)
```

Comment:
sacarla al scope global del user.js .. para evitar que se bloquee el Event Loop

Reply:
para el contexto del por que ? puede dar una ojeada 
[https://medium.com/dkatalis/eventloop-in-nodejs-ways-to-block-it-and-ways-to-avoid-b60a65bab2be](https://medium.com/dkatalis/eventloop-in-nodejs-ways-to-block-it-and-ways-to-avoid-b60a65bab2be)

Reply:
intente hacerlo antes de pullear pero me tira un error ya que las dos lineas del hash tienen que ser asyncronas. si pongo al salt en el scope global no llega a cumplir su funcion y la linea 33 tira error porque va muy rapido

Reply:
`error = new Error('data must be a string or Buffer and salt must either be a salt string or a number of rounds');`